### PR TITLE
[Enhance] Efficient implementation of PointSegClassMapping

### PR DESCRIPTION
**Pull request by Wuziyi616**
The original implementation uses for-loop to convert every label, which is super slow in large scenes (e.g. S3DIS seg dataset without sub-sampling). I modify it to use matrix indexing, which could be faster. Matrix indexing is also a common practice in other 3D Seg repos.

### CODE REVIEW

**Comment by Wuziyi616** [comment]
I have carefully checked and modified all the places which use `PointSegClassMapping` in the codebase.

File: mmdet3d/datasets/pipelines/loading.py [hunk 1]
```diff
@@ -237,10 +237,19 @@ class PointSegClassMapping(object):
...
     Args:
         valid_cat_ids (tuple[int]): A tuple of valid category.
+        max_cat_id (int): The max possible cat_id in input segmentation mask.
+            Defaults to 40.
     """
 
-    def __init__(self, valid_cat_ids):
+    def __init__(self, valid_cat_ids, max_cat_id=40):
         self.valid_cat_ids = valid_cat_ids
+        self.max_cat_id = max_cat_id
```
Comment (ZwwWayne) [inline mmdet3d/datasets/pipelines/loading.py:246]: Need to check consistency between max_cat_id and valid_cat_id

Comment (ZwwWayne) [reply mmdet3d/datasets/pipelines/loading.py:246]: also test the check in unittests

Comment (Wuziyi616) [reply mmdet3d/datasets/pipelines/loading.py]: Done.

**Review by ZwwWayne — APPROVED** [review]

**Pull request merged**